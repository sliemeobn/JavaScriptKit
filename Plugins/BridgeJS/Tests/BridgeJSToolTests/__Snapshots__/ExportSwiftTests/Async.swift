// NOTICE: This is auto-generated code by BridgeJS from JavaScriptKit,
// DO NOT EDIT.
//
// To update this file, just rebuild your project or run
// `swift package bridge-js`.

@_spi(BridgeJS) import JavaScriptKit

@_expose(wasm, "bjs_asyncReturnVoid")
@_cdecl("bjs_asyncReturnVoid")
public func _bjs_asyncReturnVoid() -> Int32 {
    #if arch(wasm32)
    let ret = JSPromise.async {
        await asyncReturnVoid()
    } .jsObject
    return _swift_js_retain(Int32(bitPattern: ret.id))
    #else
    fatalError("Only available on WebAssembly")
    #endif
}

@_expose(wasm, "bjs_asyncRoundTripInt")
@_cdecl("bjs_asyncRoundTripInt")
public func _bjs_asyncRoundTripInt(v: Int32) -> Int32 {
    #if arch(wasm32)
    let ret = JSPromise.async {
        return await asyncRoundTripInt(_: Int(v)).jsValue
    } .jsObject
    return _swift_js_retain(Int32(bitPattern: ret.id))
    #else
    fatalError("Only available on WebAssembly")
    #endif
}

@_expose(wasm, "bjs_asyncRoundTripString")
@_cdecl("bjs_asyncRoundTripString")
public func _bjs_asyncRoundTripString(vBytes: Int32, vLen: Int32) -> Int32 {
    #if arch(wasm32)
    let ret = JSPromise.async {
        let v = String(unsafeUninitializedCapacity: Int(vLen)) { b in
            _swift_js_init_memory(vBytes, b.baseAddress.unsafelyUnwrapped)
            return Int(vLen)
        }
        return await asyncRoundTripString(_: v).jsValue
    } .jsObject
    return _swift_js_retain(Int32(bitPattern: ret.id))
    #else
    fatalError("Only available on WebAssembly")
    #endif
}

@_expose(wasm, "bjs_asyncRoundTripBool")
@_cdecl("bjs_asyncRoundTripBool")
public func _bjs_asyncRoundTripBool(v: Int32) -> Int32 {
    #if arch(wasm32)
    let ret = JSPromise.async {
        return await asyncRoundTripBool(_: v == 1).jsValue
    } .jsObject
    return _swift_js_retain(Int32(bitPattern: ret.id))
    #else
    fatalError("Only available on WebAssembly")
    #endif
}

@_expose(wasm, "bjs_asyncRoundTripFloat")
@_cdecl("bjs_asyncRoundTripFloat")
public func _bjs_asyncRoundTripFloat(v: Float32) -> Int32 {
    #if arch(wasm32)
    let ret = JSPromise.async {
        return await asyncRoundTripFloat(_: v).jsValue
    } .jsObject
    return _swift_js_retain(Int32(bitPattern: ret.id))
    #else
    fatalError("Only available on WebAssembly")
    #endif
}

@_expose(wasm, "bjs_asyncRoundTripDouble")
@_cdecl("bjs_asyncRoundTripDouble")
public func _bjs_asyncRoundTripDouble(v: Float64) -> Int32 {
    #if arch(wasm32)
    let ret = JSPromise.async {
        return await asyncRoundTripDouble(_: v).jsValue
    } .jsObject
    return _swift_js_retain(Int32(bitPattern: ret.id))
    #else
    fatalError("Only available on WebAssembly")
    #endif
}

@_expose(wasm, "bjs_asyncRoundTripJSObject")
@_cdecl("bjs_asyncRoundTripJSObject")
public func _bjs_asyncRoundTripJSObject(v: Int32) -> Int32 {
    #if arch(wasm32)
    let ret = JSPromise.async {
        return await asyncRoundTripJSObject(_: JSObject(id: UInt32(bitPattern: v))).jsValue
    } .jsObject
    return _swift_js_retain(Int32(bitPattern: ret.id))
    #else
    fatalError("Only available on WebAssembly")
    #endif
}