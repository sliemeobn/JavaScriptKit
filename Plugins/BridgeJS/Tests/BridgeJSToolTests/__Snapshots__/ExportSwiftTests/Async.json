{
  "classes" : [

  ],
  "enums" : [

  ],
  "functions" : [
    {
      "abiName" : "bjs_asyncReturnVoid",
      "effects" : {
        "isAsync" : true,
        "isThrows" : false
      },
      "name" : "asyncReturnVoid",
      "parameters" : [

      ],
      "returnType" : {
        "void" : {

        }
      }
    },
    {
      "abiName" : "bjs_asyncRoundTripInt",
      "effects" : {
        "isAsync" : true,
        "isThrows" : false
      },
      "name" : "asyncRoundTripInt",
      "parameters" : [
        {
          "label" : "_",
          "name" : "v",
          "type" : {
            "int" : {

            }
          }
        }
      ],
      "returnType" : {
        "int" : {

        }
      }
    },
    {
      "abiName" : "bjs_asyncRoundTripString",
      "effects" : {
        "isAsync" : true,
        "isThrows" : false
      },
      "name" : "asyncRoundTripString",
      "parameters" : [
        {
          "label" : "_",
          "name" : "v",
          "type" : {
            "string" : {

            }
          }
        }
      ],
      "returnType" : {
        "string" : {

        }
      }
    },
    {
      "abiName" : "bjs_asyncRoundTripBool",
      "effects" : {
        "isAsync" : true,
        "isThrows" : false
      },
      "name" : "asyncRoundTripBool",
      "parameters" : [
        {
          "label" : "_",
          "name" : "v",
          "type" : {
            "bool" : {

            }
          }
        }
      ],
      "returnType" : {
        "bool" : {

        }
      }
    },
    {
      "abiName" : "bjs_asyncRoundTripFloat",
      "effects" : {
        "isAsync" : true,
        "isThrows" : false
      },
      "name" : "asyncRoundTripFloat",
      "parameters" : [
        {
          "label" : "_",
          "name" : "v",
          "type" : {
            "float" : {

            }
          }
        }
      ],
      "returnType" : {
        "float" : {

        }
      }
    },
    {
      "abiName" : "bjs_asyncRoundTripDouble",
      "effects" : {
        "isAsync" : true,
        "isThrows" : false
      },
      "name" : "asyncRoundTripDouble",
      "parameters" : [
        {
          "label" : "_",
          "name" : "v",
          "type" : {
            "double" : {

            }
          }
        }
      ],
      "returnType" : {
        "double" : {

        }
      }
    },
    {
      "abiName" : "bjs_asyncRoundTripJSObject",
      "effects" : {
        "isAsync" : true,
        "isThrows" : false
      },
      "name" : "asyncRoundTripJSObject",
      "parameters" : [
        {
          "label" : "_",
          "name" : "v",
          "type" : {
            "jsObject" : {

            }
          }
        }
      ],
      "returnType" : {
        "jsObject" : {

        }
      }
    }
  ],
  "moduleName" : "TestModule"
}